src/pkg/packager/remove.go

```diff
@@ -36,60 +36,35 @@ func (p *Packager) Remove(packageName string) error {
 	secretName := config.ZarfPackagePrefix + packageName
 	packageSecret, err := p.cluster.Kube.GetSecret(cluster.ZarfNamespace, secretName)
 	if err != nil {
-		spinner.Errorf(err, "Unable to get the secret for the package we are attempting to remove")
-
-		return err
+		return fmt.Errorf("unable to get the secret for the package we are attempting to remove: %w", err)
 	}
 
 	// Get the list of components the package had deployed
 	deployedPackage := types.DeployedPackage{}
 	err = json.Unmarshal(packageSecret.Data["data"], &deployedPackage)
 	if err != nil {
-		spinner.Errorf(err, "Unable to load the secret for the package we are attempting to remove")
-
-		return err
+		return fmt.Errorf("unable to load the secret for the package we are attempting to remove: %w", err)
 	}
 
 	// If components were provided; just remove the things we were asked to remove
 	requestedComponents := strings.Split(p.cfg.DeployOpts.Components, ",")
+	partialRemove := len(requestedComponents) > 0 && requestedComponents[0] != ""
 
-	// If components were not provided; set things up to remove all package components
-	if len(requestedComponents) < 1 || requestedComponents[0] == "" {
-		requestedComponents = []string{}
-
-		for _, component := range deployedPackage.DeployedComponents {
-			requestedComponents = append(requestedComponents, component.Name)
+	for _, c := range utils.Reverse(deployedPackage.DeployedComponents) {
+		// Only remove the component if it was requested or if we are removing the whole package
+		if partialRemove && !slices.Contains(requestedComponents, c.Name) {
+			continue
 		}
-	}
-
-	// Loop through the deployed components (in reverse order) check if they were requested and remove them if so
-	for i := len(deployedPackage.DeployedComponents) - 1; i >= 0; i-- {
-		installedComponent := deployedPackage.DeployedComponents[i]
-
-		if slices.Contains(requestedComponents, installedComponent.Name) {
-			for h := len(installedComponent.InstalledCharts) - 1; h >= 0; h-- {
-				installedChart := installedComponent.InstalledCharts[h]
-
-				spinner.Updatef("Uninstalling chart (%s) from the (%s) component", installedChart.ChartName, installedComponent.Name)
 
-				helmCfg := helm.Helm{}
-				err = helmCfg.RemoveChart(installedChart.Namespace, installedChart.ChartName, spinner)
-				if err != nil {
-					message.Errorf(err, "Unable to remove the installed helm chart (%s) from the namespace (%s) of component (%s) (were dependent components removed first?)",
-						installedChart.ChartName, installedChart.Namespace, installedComponent.Name)
-
-					return err
-				}
-
-				// Remove the uninstalled chart from the list of installed charts
-				deployedPackage.DeployedComponents[i].InstalledCharts = deployedPackage.DeployedComponents[i].InstalledCharts[:h]
-				p.updatePackageSecret(deployedPackage, secretName)
```

Comment:
This functionality does not exist in the new logic.  If a component has multiple helm charts, uninstalls the first one successfully and fails on the second, zarf's state will now be bad and will perma-fail on the now ghosted first chart.

Reply:
Pushed something out for this 👍